src/rpc/blockchain.cpp

```diff
@@ -799,16 +800,9 @@ static RPCHelpMan getblockfrompeer()
     const NodeContext& node = EnsureAnyNodeContext(request.context);
     ChainstateManager& chainman = EnsureChainman(node);
     PeerManager& peerman = EnsurePeerman(node);
-    CConnman& connman = EnsureConnman(node);
 
-    uint256 hash(ParseHashV(request.params[0], "hash"));
-
-    const NodeId nodeid = static_cast<NodeId>(request.params[1].get_int64());
-
-    // Check that the peer with nodeid exists
-    if (!connman.ForNode(nodeid, [](CNode* node) {return true;})) {
-        throw JSONRPCError(RPC_MISC_ERROR, strprintf("Peer nodeid %d does not exist", nodeid));
-    }
```

Comment:
Why are you removing this? Where is the peer id checked now?

Reply:
It's checked in `FetchBlock`:
https://github.com/bitcoin/bitcoin/blob/ac92ab6da58e34993d0641b98eef5b5f55b6cbf9/src/net_processing.cpp#L1436-L1438

As suggested here:
https://github.com/bitcoin/bitcoin/pull/20295#discussion_r764726512

Reply:
But this check was for the case where FetchBlock is not called (block already downloaded)

Reply:
I see. But we only return one error, and it seems to be me that the "block already downloaded" error is more relevant anyway.

Reply:
That's currently just a warning. Maybe it should be an error?

Reply:
Good point, let's just make it an error, so it's easier to reason about.

Reply:
my previous test:
```
$ ./src/bitcoin-cli -signet getblockfrompeer 00000152031b708d8b29013d55aedd24e8db90af7fcf7f1e51881ad97cf0e9a4 0
{
  "warnings": "Block already downloaded"
}
```
now
```
$ ./src/bitcoin-cli -signet getblockfrompeer 00000152031b708d8b29013d55aedd24e8db90af7fcf7f1e51881ad97cf0e9a4 0
error code: -1
error message:
Block already downloaded
```